File: engine/model/src/main/scala/tools/packet/message/NotifyJobAdvance.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class NotifyJobAdvance(private var _type: Int, private var _job: Int, private var _characterName: String) extends PacketInput {
  def theType: Int = _type

  def job: Int = _job

  def characterName: String = _characterName

  override def opcode(): SendOpcode = SendOpcode.NOTIFY_JOB_CHANGE
}

File: engine/engine-model/src/main/java/tools/packet/message/NotifyJobAdvance.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record NotifyJobAdvance(Integer theType, Integer job, String characterName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NOTIFY_JOB_CHANGE;
   }
}